pkg/asset/manifests/azure/cluster.go

```diff
@@ -92,20 +84,67 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 		nodeSubnetID = "UNKNOWN"
 	}
 
+	apiServerLB := capz.LoadBalancerSpec{
+		Name: fmt.Sprintf("%s-internal", clusterID.InfraID),
+		BackendPool: capz.BackendPool{
+			Name: fmt.Sprintf("%s-internal", clusterID.InfraID),
+		},
+		LoadBalancerClassSpec: capz.LoadBalancerClassSpec{
+			Type: capz.Internal,
+		},
+	}
+
+	controlPlaneOutboundLB := &capz.LoadBalancerSpec{
+		Name:             clusterID.InfraID,
+		FrontendIPsCount: to.Ptr(int32(1)),
+	}
+	if installConfig.Config.Platform.Azure.OutboundType == azure.UserDefinedRoutingOutboundType {
+		controlPlaneOutboundLB = nil
+	}
+
 	virtualNetworkID := ""
 	if installConfig.Config.Azure.VirtualNetwork != "" {
 		client, err := installConfig.Azure.Client()
 		if err != nil {
 			return nil, fmt.Errorf("failed to get azure client: %w", err)
 		}
+		ctx := context.TODO()
 
-		virtualNetwork, err := client.GetVirtualNetwork(context.TODO(), installConfig.Config.Azure.NetworkResourceGroupName, installConfig.Config.Azure.VirtualNetwork)
+		virtualNetwork, err := client.GetVirtualNetwork(ctx, installConfig.Config.Azure.NetworkResourceGroupName, installConfig.Config.Azure.VirtualNetwork)
 		if err != nil {
 			return nil, fmt.Errorf("failed to get azure virtual network: %w", err)
 		}
 		if virtualNetwork != nil {
 			virtualNetworkID = *virtualNetwork.ID
 		}
+		lbip := "10.0.0.100"
```

Comment:
Nit: Would be nice to move this extended logic for the creation of internalLB to a method of its own.